### ISSUE TITLE
Workshop feedback

### ISSUE THREAD

**Issue by Vidarls** [bjartnes/http-workshop#20]
# What did you learn?

💡 I learned... ?
A nice way to quickly respond to http requests without setting up a test server - superb for testing.
💡 And maybe more...?
A bit more on the general HTTP syntax

# What will you try to use in your project/next project?
⚙Thing I will try
Using nc for testing both as client and server.
⚙Another thing I will try.
K6 for loadtesting

# What was good about the format, what should we do more of?
✔I liked...
The prepared test setup, with docker and codespace options. Took away a lot of friction in the workshop
✔And I would like to do more of...

# What was not so good... Do less of... Other things to improve
😢This made me sad
💔This dissapointed me
🥱💤 This was boring
For me the pace was a little bit slow, I would have enjoyed som more esoteric cases and gotchas - but it is the struggle with the workshop format. Kudos for providing the extended assignments as compensation for this.

Tl;dr: Maybe a bit much time for the content, but I enjoyed the hands on concrete format.

**Comment by alinjie** [comment]
Cool workshop! Good to learn how to use these tools and how HTTP works under the hood! Thanks!

**Comment by caring-coder** [comment]
> # What did you learn?
I learned how easy it is to actually manipulate http and tcp with simple cli tools.

> # What will you try to use in your project/next project?
I will definitely make some workshop to increase practical knoledge of my colleagues about network protocols, i will also take a look around with K6 and compare it with the Gatling framework that i was using.

> # What was good about the format, what should we do more of?
I liked the prepared test setup, clearly, and also the written clear and detailed assignments.
I think an optional Part 0 with ip or udp manipulation might be a plus, to build from deeper foundations.
To have a public server that we can try to connect to or answer to (extreme startup kata style) graphically showing the reception of request might be a dynamic and stimulating way to show the progress of the session.
 
> # What was not so good... Do less of... Other things to improve
The pace was a bit all over the place, sometimes too fast, sometimes really slow.
Maybe make break time more clear.
I think making pairing mandatory might make everyone be more engaged and focused, and also to let each other unblock themselves without depending too much on you.

**Comment by bjorn-einar-bjartnes-4ss** [comment]
@josephverron Could you elaborate a little bit on what you would like to take away from a part 0, provide some ideas or examples? Would love to add more stuff, in a part x or 0 og infinity :) Just need to understand it a little better....

**Comment by caring-coder** [comment]
@bjorn-einar-bjartnes-4ss 

The part 0 i imagine would be akin to a similar to an optional "TCP by hand" workshop i guess.

Something like trying to get a TCP connection up and running over IP, from manual interaction until having a tool abstracting it away (emulating netcat, in a sense), and then move on to the HTTP part of the workshop.

The goal would be to let learners connect with the deeper layer of internet protocol, like understanding packet-size related issues and what nots ...

**Comment by bjorn-einar-bjartnes-4ss** [comment]
Get it. I have some material and some ideas. Perhaps something with setting up som machines and vnets in Azure with bicep etc to learn more about networks and send ip packetc etc. Perhaps this could be a separate repo with a focus on layers below tcp/ip? And we link to that for those who want to go deeper?
